geniza/annotations/views.py

```diff
@@ -0,0 +1,90 @@
+import json
+
+from django.http import HttpResponse, JsonResponse
+from django.shortcuts import render
+from django.utils.decorators import method_decorator
+from django.views.decorators.csrf import csrf_exempt
+from django.views.decorators.http import require_http_methods
+from django.views.generic.base import View
+from django.views.generic.detail import SingleObjectMixin
+from django.views.generic.list import MultipleObjectMixin
+
+from geniza.annotations.models import Annotation
+
+
+class AnnotationResponse(JsonResponse):
+    content_type = 'application/ld+json; profile="http://www.w3.org/ns/anno.jsonld"'
+
+
+@method_decorator(csrf_exempt, name="dispatch")  # disable csrf for testing with curl
+class AnnotationList(View, MultipleObjectMixin):
+    model = Annotation
+    http_method_names = ["get", "post"]
+
+    paginate_by = None  # disable pagination for now
+
+    def get(self, request, *args, **kwargs):
+        # strictly speaking, annotations endpoint should return an annotation container,
+        # but that structure looks terrible to work with and we need a way to search,
+        # which is not defined in the w3c annotation protocol.
+
+        # implement something similar to SAS search by uri
+        annotations = self.get_queryset()
+        # if a target uri is specified, filter annotations
+        target_uri = self.request.GET.get("target_uri")
+        if target_uri:
+            annotations = annotations.filter(content__target__source__id=target_uri)
+
+        # return json response with list of annotations
+        # TODO: eventually we'll need to limit or paginate this
```

Comment:
Probably need UI for that too…

Reply:
how do you mean, UI?

Reply:
I was thinking you meant limit or paginate the results for UI _reasons_, i.e. the list is too long to display, but now I realize you probably just mean to avoid the payload size from getting to large.

Reply:
Ah, yes. Right now the main annotations endpoint is returning everything, which I thought would be helpful as we're developing it. Once we split the tei into chunks and convert to annotations, that will be far too many to display in a single result. But was thinking we could do something simple to display counts and page links or similar.